xbmc/addons/AddonManager.cpp

```diff
@@ -1025,4 +1050,41 @@ void CAddonMgr::FindAddons(ADDON_INFO_LIST& addonmap, const std::string& path)
   }
 }
 
+bool CAddonMgr::IsFromTrustedRepo(const AddonPtr& addon) const
+{
+  for (const auto& trustedRepo : m_trustedAddonRepos)
+  {
+    if (addon->Origin() == trustedRepo)
+      return true;
+  }
+
+  return false;
+}
+
+void CAddonMgr::AddAddonIfLatest(const AddonPtr& addonToAdd,
+                                 std::map<std::string, AddonPtr>& map) const
+{
+  const auto& latest_known = map.find(addonToAdd->ID());
+  if (latest_known == map.end() || addonToAdd->Version() > latest_known->second->Version())
+    map[addonToAdd->ID()] = std::move(addonToAdd);
```

Comment:
```suggestion
    map[addonToAdd->ID()] = addonToAdd;
```
You won't be ablle to move anything out of a const reference